### PR TITLE
Fix flaky tests in CredentialsManagerSpec

### PR DESCRIPTION
**Pull request by Widcket**
### Changes

There were a couple of flaky tests due to the fact that the async assertions were not being made inside a `waitUntil(timeout: Timeout)` closure, with the respective call to `done()`.

Also, a couple of test cases that were superfluous were removed, as they were essentially duplicates.

### Testing

* [ ] This change adds unit test coverage
* [ ] This change has been tested on the latest version of the platform/language or why not

### Checklist

* [ ] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [ ] All existing and new tests complete without errors
* [ ] All active GitHub checks have passed

### CODE REVIEW

File: Auth0Tests/CredentialsManagerSpec.swift [hunk 1]
```diff
@@ -370,28 +370,40 @@ class CredentialsManagerSpec: QuickSpec {
...
 
             afterEach {
-                A0SimpleKeychain().clearAll()
+                _ = credentialsManager.clear()
             }
 
             it("should error when no credentials stored") {
-                A0SimpleKeychain().clearAll()
-                credentialsManager.credentials { result in
-                    expect(result).to(haveCredentialsManagerError(CredentialsManagerError(code: .noCredentials)))
```
Comment (Widcket) [inline Auth0Tests/CredentialsManagerSpec.swift:379]: There was no `waitUntil(timeout: Timeout)`.

File: Auth0Tests/CredentialsManagerSpec.swift [hunk 2]
```diff
@@ -644,27 +639,6 @@ class CredentialsManagerSpec: QuickSpec {
 
             context("serial renew from different threads") {
 
-                it("should not renew the credentials after the previous renewal operation succeeded") {
```
Comment (Widcket) [inline Auth0Tests/CredentialsManagerSpec.swift:647]: Same as above.

**Review by adamjmcgrath — APPROVED** [review]
